File: src/packets/PacketC2SZonePower.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x08
 * Zone Power
 * Turns a Zone On/Off
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unsigned Char) Zone ID
 *      (Unsigned Char) On/Off State
 */
class PacketC2SZonePower extends PacketC2S {
    parseData() {
        this.ctrllrID = this._buffer.readUInt8();
        this.zoneID = this._buffer.readUInt8();
        this.power = (this._buffer.readUInt8() == 1);
    }
}

module.exports = PacketC2SZonePower;

File: src/RNetPi.Core/Packets/PacketC2SZonePower.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x08
/// Zone Power
/// Turns a Zone On/Off
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (Unsigned Char) On/Off State
/// </summary>
public class PacketC2SZonePower : PacketC2S
{
    public const byte ID = 0x08;

    public byte ControllerID { get; private set; }
    public byte ZoneID { get; private set; }
    public bool Powered { get; private set; }

    public PacketC2SZonePower(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        ControllerID = Reader.ReadByte();
        ZoneID = Reader.ReadByte();
        Powered = Reader.ReadByte() == 1;
    }
}